src/state/reducers/month-state/schedule-data/schedule-data.action-creator.ts

```diff
@@ -35,21 +36,48 @@ export class ScheduleDataActionCreator {
     };
   }
 
-  static setScheduleFromMonthDM(newMonth: MonthDataModel): ThunkFunction<ScheduleDataModel> {
+  static setScheduleFromMonthDM(
+    newMonth: MonthDataModel,
+    saveInDatabase = true,
+    revision?: RevisionType
+  ): ThunkFunction<ScheduleDataModel> {
     return async (dispatch, getState): Promise<void> => {
-      const history = getState().history;
-      const [prevMonth, nextMonth] = await getMonthNeighbours(newMonth, history);
+      if (_.isNil(revision)) {
+        revision = getState().actualState.revision;
+      }
+      const [prevMonth, nextMonth] = await new LocalStorageProvider().fetchOrCreateMonthNeighbours(
+        newMonth,
+        revision
+      );
       const newSchedule = extendMonthDMToScheduleDM(prevMonth, newMonth, nextMonth);
-      await this.setScheduleFromScheduleDM(newSchedule)(dispatch, getState);
+      await this.setScheduleFromScheduleDM(newSchedule, saveInDatabase)(dispatch, getState);
     };
   }
 
-  static setScheduleFromKeyIfExists(monthKey: ScheduleKey): ThunkFunction<ScheduleDataModel> {
+  static setScheduleFromKeyIfExistsInDB(
+    monthKey: ScheduleKey,
+    revision?: RevisionType,
+    baseMonthModel?: MonthDataModel
+  ): ThunkFunction<ScheduleDataModel> {
     return async (dispatch, getState): Promise<void> => {
-      const history = getState().history;
-      const monthDataModel = await fetchMonthDM(monthKey, history);
+      let monthDataModel;
+      if (_.isNil(revision)) {
+        revision = getState().actualState.revision;
+      }
+      if (baseMonthModel) {
+        monthDataModel = await new LocalStorageProvider().fetchOrCreateMonthRevision(
+          monthKey,
+          revision,
+          baseMonthModel
+        );
+      } else {
+        monthDataModel = await new LocalStorageProvider().getMonthRevision(
+          monthKey.getRevisionKey(revision)
+        );
+      }
+
       if (!_.isNil(monthDataModel)) {
-        dispatch(this.setScheduleFromMonthDM(monthDataModel));
+        dispatch(this.setScheduleFromMonthDM(monthDataModel, false, revision));
       }
```

Comment:
jeśli oba awaity nie mogą zwrócić nila, to tu nie może być nil i nie trzeba sprawdzać ( nie sprawdzałem tych funkcji co zwracają)

Reply:
ale ogólnie tutaj jest tak kod wypchany takimi guardami jeśli czegoś nie ma, że trochę za dużo, może da się przemyśleć czy wsszystkie są potrzebne?

Reply:
Faktycznie nie jest to najpięknieszy kod, ale nie za bardzo mogę to ugryźć inaczej. Jeden z awaitów zwrca undefined dlatego jest ten guard. 
Jedyne rozwiązanie jakie mi przychodzi do głowy to rozbicie tego na 2 funkcje, jedna gdzie trzeba podać baseMonthModel, a druga bez tego, tylko, że mamy trochę powtarzającej się logiki i tego chciałem uniknąć. 

Pytanie czy lepiej mieć więcej kodu, ale lepiej opisanego czy dodać takiego guarda